### ISSUE TITLE
No payouts for 2 days

### ISSUE THREAD

**Issue by vmajor** [SChernykh/p2pool#88]
I was receiving payouts every 40-60 minutes , but nothing at all for the past two days.

Daemon is running and is in sync
p2pool is running
xmrig is running

I even have BLOCK FOUND message in p2pool output. But no actual reward, no payout.

![image](https://user-images.githubusercontent.com/5185905/147119381-6f83dc23-c308-40ff-9985-8c876620d08c.png)

Please help me troubleshoot this

**Comment by SChernykh** [comment]
Show the output of `status` command in p2pool console. You're probably out of sync with other p2pool nodes. Also please check what https://p2pool.observer/ shows for your wallet address.

**Comment by vmajor** [comment]
Status:
![image](https://user-images.githubusercontent.com/5185905/147189008-7dc2b7aa-b744-48e1-8835-f02bbdd568d7.png)

Observer:
![image](https://user-images.githubusercontent.com/5185905/147189143-323fd516-daa5-4140-be77-4d3ae8d5941b.png)

**Comment by SChernykh** [comment]
You have no connections on the stratum server, so check where your xmrig is pointed to and if it shows any errors.

**Comment by vmajor** [comment]
How can I check where it is pointed to? There are no errors and it is being given jobs by p2pool:

![image](https://user-images.githubusercontent.com/5185905/147213200-1ba179d2-822e-4f5f-a70d-c47a1d3c2f10.png)

**Comment by SChernykh** [comment]
Do you run xmrig and p2pool on the same machine? Check p2pool status again. Also try to set fixed difficulty on xmrig to see your hashrate on p2pool side. Also, it will take you ~1 day 17 hours on average to get a share with this hashrate.

**Comment by vmajor** [comment]
Yes, everything (daemon, p2pool and xmrig) are on localhost. I limited the hash rate to 6000, and p2pool status shows this. What does this tell me?:

![image](https://user-images.githubusercontent.com/5185905/147219124-d650fcac-3d4d-419c-ae68-cf6ef468d27f.png)


xmrig:
![image](https://user-images.githubusercontent.com/5185905/147219239-05ad3753-903b-4f1f-8e9d-23689d82f51f.png)

**Comment by SChernykh** [comment]
Everything is working, but it's better to set fixed difficulty to 300000. But it's impossible to "receive payouts every 40-60 minutes" with this hashrate. You'll be finding pool shares every 1-3 days.

**Comment by trasherdk** [comment]
Is it possible to do some kind of calculation, to output, in that log, something to estimate a expectation, based on hashrate?

I understand the concept PoW, and random X, and the level of luck involved, but for the sake of "low powered" miners,
who don't understand, who keep asking "My laptop is glowing, mining for 24 hours, but I still haven't been paid", answering that kinda questions.

**Comment by SChernykh** [comment]
> Is it possible to do some kind of calculation, to output, in that log, something to estimate a expectation, based on hashrate?
> 
> I understand the concept PoW, and random X, and the level of luck involved, but for the sake of "low powered" miners, who don't understand, who keep asking "My laptop is glowing, mining for 24 hours, but I still haven't been paid", answering that kinda questions.

It's already in https://p2pool.io/#faq I don't want to litter p2pool console messages with ELI5 explanations.

**Comment by vmajor** [comment]
The reason why I posted this question was because this was my experience until 3 days ago:

![image](https://user-images.githubusercontent.com/5185905/147279020-f321a18b-83ac-45af-9fc6-87c1e2fc280f.png)

There have been no payouts since, but as you said I will keep it running and see what happens by tomorrow.

**Comment by SChernykh** [comment]
You get many payouts from each found share on the main p2pool, all these payouts 3 days ago could've been from a single share.

**Comment by vmajor** [comment]
OK. I just found this again in p2pool, and there are no payouts. Please let me know if this is normal and expected behaviour:

![image](https://user-images.githubusercontent.com/5185905/147313889-35ce52f9-eaef-4db3-afbb-f21688ac135c.png)

**Comment by SChernykh** [comment]
It's a block found by p2pool, not a share found by you. Check p2pool status in console or check https://p2pool.observer/ to see if you found any shares.

**Comment by vmajor** [comment]
Well, it is not encouraging. Should I be worried that it cannot see my hash rate?

![image](https://user-images.githubusercontent.com/5185905/147340462-85a80a46-64c6-4845-a35e-495d13d85acf.png)

**Comment by SChernykh** [comment]
You will see non-zero number on the observer site when you find a share.

**Comment by tywkah** [comment]
> It's a block found by p2pool, not a share found by you. Check p2pool status in console or check https://p2pool.observer/ to see if you found any shares.

and again, as you see, [it's confusing](https://github.com/SChernykh/p2pool/issues/67#issuecomment-957970776)
(the 2nd part was beaten by scripting and timeout with the firewall)

**Comment by SChernykh** [comment]
`BLOCK FOUND` message is exactly what it says. It's not confusing. Everything you need to know to not be confused is aleady described in https://github.com/SChernykh/p2pool#how-pplns-works-in-p2pool and in https://p2pool.io/#faq

**Comment by tywkah** [comment]
OK, what if someone wanna see only his blocks (i mean only in pplns share that would be payed out) in the console output?

**Comment by SChernykh** [comment]
For that, someone just needs to check his Monero wallet because blocks with pplns shares pay out to it.

**Comment by vmajor** [comment]
I started receiving/received payouts again to my wallet. Thus everything again appears to be normal and as you said S.

The cognitive confusion is with the concept of "pool" and still having to meet other conditions in order for block discovery to be rewarded.
